service/history/hsm/events.go

```diff
@@ -0,0 +1,33 @@
+// The MIT License
+//
+// Copyright (c) 2024 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package hsm
+
+import enumspb "go.temporal.io/api/enums/v1"
+
+// EventDefinition is a definition for a history event for a given event type.
+type EventDefinition interface {
+	Type() enumspb.EventType
+	// IsWorkflowTaskTrigger returns a boolean indicating whether this event type should trigger a workflow task.
+	IsWorkflowTaskTrigger() bool
```

Comment:
hmm workflowTask is a workflow ASM specific concept... so don't think we should register this interface to the hsm framework. I remember we added some registry to the workflow package some time ago? Is it possible to reuse that?

Reply:
Oh and if the registry was on the workflow package, I'd need to put this code in a different package to avoid circular dependencies.

Reply:
> history event apply hooks
hmm maybe. But I guess I want explore some other possibilities as well.
 
I am not concerned about the interface, but the WorkflowTask concept in the interface. Another way I can think of is have a ScheduleWorkflowTask method on mutable state and the task executor just call that method at the end of the a Write Access. That also decouples event type from workflow task creation (e.g. sometimes a WorkflowExecutionSignaledEvent doesn't generate a workflow task)